src/diagnostics.rs

```diff
@@ -279,18 +279,19 @@ fn report_msg<'tcx>(
         DiagLevel::Note => tcx.sess.diagnostic().span_note_diag(span, title),
     };
 
+    let first_line = span_msg.get(0).cloned().unwrap_or_default();
```

Comment:
Why single out the first line?

You can just add the `err.note("(no span available)");` after all of them IMO, keeping the code simpler.

Reply:
:facepalm: